### PR TITLE
feat(hosting): use common offers selector

### PR DESCRIPTION
**Pull request by mohammed-zahaf**
| Question         | Answer
| ---------------- | ---
| Branch?          | `release/web-nwo`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-11028
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
1 - refactoring to use common hosting domain component
2 - use common offers components

### CODE REVIEW

File: packages/manager/apps/web/client/app/domain/webhosting/order/domain-webhosting-order.routing.js [hunk 1]
```diff
@@ -7,8 +7,17 @@ const resolve = {
   /* @ngInject */
   getAvailableModules: (cartId, WebHostingOrder) => (offer) =>
     WebHostingOrder.getAvailableModules(cartId, offer),
-  availableOffers: /* @ngInject */ (cartId, user, WebHostingOrder) =>
-    WebHostingOrder.getAvailableOffers(cartId, user.ovhSubsidiary),
+
+  /** TODO: for the moment the available offers must be hardcoded. Later they will be dynamic */
+  availableOffers: /* @ngInject */ () => [
```
Comment (darsene) [inline packages/manager/apps/web/client/app/domain/webhosting/order/domain-webhosting-order.routing.js:12]: Maybe, you should move this part in a service even it's hard-coded for now. In this way, the routing part won't have to be modified when the offers will be "dynamic". Only the service logic will be change. What do you think ?

File: packages/manager/apps/web/client/app/domain/webhosting/order/domain-webhosting-order.routing.js [hunk 2]
```diff
@@ -109,6 +118,13 @@ export default /* @ngInject */ ($stateProvider) => {
         ...resolve,
         goBackToDashboard: /* @ngInject */ ($state) => () =>
           $state.go('app.domain.product.information'),
+
+        user: /* @ngInject */ (coreConfig) => coreConfig.getUser(),
+
+        catalog: /* @ngInject */ (user, WebHostingOrder) => {
+          console.log('ZM:: Order.routing.catalog', user);
```
Comment (darsene) [inline packages/manager/apps/web/client/app/domain/webhosting/order/domain-webhosting-order.routing.js:125]: to remove I guess

Comment (mohammed-zahaf) [reply packages/manager/apps/web/client/app/domain/webhosting/order/domain-webhosting-order.routing.js:125]: my bad forget, thanks

File: packages/manager/apps/web/client/app/domain/webhosting/order/domain-webhosting-order.routing.js (hunk 1 above)
Comment (mohammed-zahaf) [reply packages/manager/apps/web/client/app/domain/webhosting/order/domain-webhosting-order.routing.js:12]: why not, why i don't do it ?
currently the service already implement the available offers function.
But it return the no desired offers, so i guess once we have the required offers, we can reuse the updated endpoint.

Let me know

**Review by sachinrameshn — APPROVED** [review]

File: packages/manager/apps/web/client/app/domain/webhosting/order/domain-webhosting-order.routing.js (hunk 1 above)
Comment (darsene) [reply packages/manager/apps/web/client/app/domain/webhosting/order/domain-webhosting-order.routing.js:12]: No problem, it's ok for me

**Review by darsene — APPROVED** [review]

**Pull request merged**